pkg/model/stack_serializer.go

```diff
@@ -597,6 +602,25 @@ func getSvcPorts(public bool, rawPorts, rawExpose []PortRaw) (bool, []Port, erro
 	return public, ports, nil
 }
 
+// translateOktetoStacksPortsIntoComposeSyntax translates okteto stack port syntax into compose ports syntax
+// We need this translation because we are using the same serializer for okteto stacks and docker compose syntax
+// Otherwise we don't know which ports should create an endpoint and which ones not
+// Example 1:
+// Stacks syntax  -> Compose syntax
+// ports:         | ports:
+//   - 5000       |   - 5000:5000
+//   - 3000       |   - 3000:3000
+// public: true   |
+func translateOktetoStacksPortsIntoComposeSyntax(ports []PortRaw) []PortRaw {
```

Comment:
I prefer to name it as translate, because we are not filling every port, we are just filling if is an okteto stack and the service is set to true, so we need to translate the ports to a compose syntax